rmw_connext_xtypes_dynamic_cpp/src/functions.cpp

```diff
@@ -889,16 +1095,76 @@ rmw_take_response(
   const rmw_client_t * client,
   void * ros_request_header, void * ros_response, bool * taken)
 {
-  *taken = false;
-  return RMW_RET_ERROR;
+  if (taken == NULL) {
+    rmw_set_error_string("taken argument can't be null");
+    return RMW_RET_ERROR;
+  }
+
+  if (client->implementation_identifier != rti_connext_dynamic_identifier) {
+    rmw_set_error_string("client handle not from this implementation");
+    // printf("but from: %s\n", client->implementation_identifier);
+    return RMW_RET_ERROR;
+  }
+
+  ConnextDynamicClientInfo * client_info = \
+    static_cast<ConnextDynamicClientInfo *>(client->data);
+  connext::Requester<DDS_DynamicData, DDS_DynamicData> * requester = client_info->requester_;
+
+  connext::LoanedSamples<DDS::DynamicData> replies = requester->take_replies(1);
+  typedef connext::LoanedSamples<DDS::DynamicData>::iterator iterator;
+  for (iterator it = replies.begin(); it != replies.end(); ++it) {
```

Comment:
You could use the for range loop or even just `auto it = replies.begin()`. It's not important to change this one, but we are using C++11, we should make the most of it!